### PR TITLE
[AOTI] Switch to use shim v2

### PR DESCRIPTION
**Pull request by hl475**
Differential Revision: D56709309

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/config.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def is_fbcode():
 )
 
 c_shim_version = os.environ.get(
-    "TORCHINDUCTOR_C_SHIM_VERSION", "1" if is_fbcode() else "2"
+    "TORCHINDUCTOR_C_SHIM_VERSION", "1" if (is_fbcode() and torch.version.hip) else "2"
```
Comment (desertfire) [inline torch/_inductor/config.py:48]: We should get rid of `torch.version.hip`, but let's chat offline to figure out how to do it.

**Review by desertfire — APPROVED** [review]

**Comment by clee2000** [comment]
@pytorchot revert -m "tests failed internally D56709309" -c ghfirst

**Pull request closed without merging**